### PR TITLE
[misc] add a linter that automatically checks dead links

### PR DESCRIPTION
**Pull request by junaire**
Signed-off-by: Jun Zhang <jun@junz.org>

### CODE REVIEW

File: .github/workflows/dead_link_checker.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        use-quiet-mode: 'yes'
+        use-verbose-mode: 'yes'
+        check-modified-files-only: 'yes'
+        config-file: '.github/workflows/mdcheck.json'
+        max-depth: 2
```
Comment (inclyc) [inline .github/workflows/dead_link_checker.yml]: why 2? (we could have nested & very deep file structure)

**Review by inclyc — APPROVED** [review]
LG

**Comment by inclyc** [comment]
Mergify did not work here (?

**Comment by junaire** [comment]
> Mergify did not work here (?

It's at least not handling the request in time, maybe it just got some delays idk...

**Pull request merged**